app/src/main/java/org/covidwatch/android/presentation/home/HomeViewModel.kt

```diff
@@ -96,12 +96,11 @@ class HomeViewModel(
         }
     }
 
-    fun onRefreshRequested(lifecycle: LifecycleOwner) {
-        contactEventsDownloader.executePublicContactEventsRefresh(lifecycle) { success -> onRefreshFinished(success) }
-    }
-
-    private fun onRefreshFinished(success: Boolean) {
-        _isRefreshing.value = false
+    fun onRefreshRequested() {
+        val state = contactEventsDownloader.executePublicContactEventsRefresh()
```

Comment:
I like this idea! Way cleaner than I proposed :+1: 
However, I believe we should remove the source every time.
So it will be something like this:
```
_isRefreshing.value = false
_isRefreshing.removeSourcce(state)
```

`_isRefreshing.value = false` Considering in `executePublicContactEventsRefresh`, you return LiveData directly from `getWorkInfosForUniqueWorkLiveData`

Reply:
Yeah, I was wondering about that.